core/sr-primitives/src/traits.rs

```diff
@@ -786,3 +787,12 @@ pub trait RuntimeApiInfo {
 	/// The version of the runtime api.
 	const VERSION: u32;
 }
+
+/// Something that can validate unsigned extrinsics.
+pub trait ValidateUnsigned {
+	/// The call to validate
+	type Call;
+
+	/// Return the validity of the call
+	fn validate_unsigned(call: &Self::Call) -> TransactionValidity;
```

Comment:
Why is this not just an `Extrinsic`?

Reply:
in the current implementation a module implement validated_unsigned for its own module::Call.

and construct runtime does a match on the outer_call and call validated_unsigned for the variant specified.

We could change by making things Into<Call> and implement outer_call or even extrinsic as such. but all in all what a module can validate will be the Call he specified, not the extrinsic.

Reply:
hmm at the same time a Runtime may want to implement validate_unsigned for something else than a Call.